### PR TITLE
Use http.Header instead of custom type

### PR DESCRIPTION
**Pull request by wendigo**
Get rid of TODO and custom Header type.

### CODE REVIEW

**Comment by wendigo** [comment]
Actually that was only todo in the source-code and now it's gone :sunglasses:

File: caddyhttp/header/header.go [hunk 1]
```diff
@@ -44,16 +49,9 @@ func (h Headers) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 
 type (
 	// Rule groups a slice of HTTP headers by a URL pattern.
-	// TODO: use http.Header type instead?
```
Comment (mholt) [inline caddyhttp/header/header.go]: YES, I love getting rid of TODOs! Thanks!

**Comment by mholt** [comment]
Oops. I merged #1212 before this which created a conflict here. Sync up and we can merge it!

File: caddyhttp/header/header.go (hunk 1 above)
Comment (wendigo) [inline caddyhttp/header/header.go]: Same here. Done - ready to merge

**Comment by mholt** [comment]
Thank you @wendigo! :tada:

**Pull request merged**